keylime/config.py

```diff
@@ -73,45 +74,257 @@ def environ_bool(env_name, default):
     # Different default WORK_DIR in TEST_MODE
     WORK_DIR = os.getenv("KEYLIME_DIR", os.getcwd())
 
-# Config files can be merged together, reading from the system to the
-# user.
-CONFIG_FILES = ["/usr/etc/keylime.conf", "/etc/keylime.conf", os.path.expanduser("~/.config/keylime.conf")]
-if "KEYLIME_CONFIG" in os.environ:
-    CONFIG_FILES.insert(0, os.environ["KEYLIME_CONFIG"])
-
-
-def get_config() -> configparser.RawConfigParser:
-    """Read configuration files and merge them together."""
-    if not getattr(get_config, "config", None):
+# Possible paths for base configuration files
+CONFIG_FILES = {
+    "agent": ["/etc/keylime/agent.conf", "/usr/etc/keylime/agent.conf"],
+    "verifier": ["/etc/keylime/verifier.conf", "/usr/etc/keylime/verifier.conf"],
+    "tenant": ["/etc/keylime/tenant.conf", "/usr/etc/keylime/tenant.conf"],
+    "registrar": ["/etc/keylime/registrar.conf", "/usr/etc/keylime/registrar.conf"],
+    "ca": ["/etc/keylime/ca.conf", "/usr/etc/keylime/ca.conf"],
+    "logging": ["/etc/keylime/logging.conf", "/usr/etc/keylime/logging.conf"],
+}
+
+# Paths to directories in which options can be overriden using configuration
+# snippets
+CONFIG_SNIPPETS_DIRS = {
+    "agent": ["/etc/keylime/agent.conf.d", "/usr/etc/keylime/agent.conf.d"],
```

Comment:
Technically "/usr/etc/keylime/*.conf.d" can be skipped, as the user can only deploy to "/etc/keylime/*.conf.d". But this is creating a very nice place for the distributor to put config snippet that does not come from upstream. I like it.

Reply:
Great, It was one of the goals, to have a easy way for the packager to add snippets on top of the default configuration.